pkg/solana/logpoller/observability.go

```diff
@@ -0,0 +1,179 @@
+// It doesn't change internal logic, because all calls are delegated to the origin ORM
+type ObservedORM struct {
+	ORM
+	queryDuration *prometheus.HistogramVec
+	datasetSize   *prometheus.GaugeVec
+	logsInserted  *prometheus.CounterVec
+	chainID       string
+}
+
+var _ ORM = &ObservedORM{}
+
+// NewObservedORM creates an observed version of log poller's ORM created by NewORM
+// Please see ObservedLogPoller for more details on how latencies are measured
+func NewObservedORM(chainID string, ds sqlutil.DataSource, lggr logger.Logger) *ObservedORM {
+	return &ObservedORM{
+		ORM:           NewORM(chainID, ds, lggr),
+		queryDuration: lpQueryDuration,
+		datasetSize:   lpQueryDataSets,
+		logsInserted:  lpLogsInserted,
+		chainID:       chainID,
+	}
+}
+
+func (o *ObservedORM) InsertLogs(ctx context.Context, logs []Log) error {
+	err := withObservedExec(o, "InsertLogs", create, func() error {
+		return o.ORM.InsertLogs(ctx, logs)
+	})
+	trackInsertedLogs(o, logs, err)
+	return err
+}
+
+func (o *ObservedORM) InsertFilter(ctx context.Context, filter Filter) (id int64, err error) {
+	return id, withObservedExec(o, "InsertFilter", create, func() (err error) {
+		id, err = o.ORM.InsertFilter(ctx, filter)
+		return err
+	})
+}
+
+func (o *ObservedORM) SelectFilters(ctx context.Context) ([]Filter, error) {
+	return withObservedQuery(o, "SelectFilters", func() ([]Filter, error) {
+		return o.ORM.SelectFilters(ctx)
+	})
+}
+
+func (o *ObservedORM) DeleteFilters(ctx context.Context, filters map[int64]Filter) error {
+	return withObservedExec(o, "DeleteFilters", del, func() error {
+		return o.ORM.DeleteFilters(ctx, filters)
+	})
+}
+
+func (o *ObservedORM) MarkFilterDeleted(ctx context.Context, id int64) error {
+	return withObservedExec(o, "MarkFilterDeleted", del, func() error {
+		return o.ORM.MarkFilterDeleted(ctx, id)
+	})
+}
+
+func (o *ObservedORM) MarkFilterBackfilled(ctx context.Context, id int64) error {
+	return withObservedExec(o, "MarkFilterBackfilled", create, func() error {
+		return o.ORM.MarkFilterBackfilled(ctx, id)
+	})
+}
+
+func (o *ObservedORM) SelectSeqNums(ctx context.Context) (map[int64]int64, error) {
+	return withObservedQuery(o, "SelectSeqNums", func() (map[int64]int64, error) {
+		return o.ORM.SelectSeqNums(ctx)
+	})
+}
+
+func (o *ObservedORM) FilteredLogs(ctx context.Context, filter []query.Expression, limitAndSort query.LimitAndSort, queryName string) ([]Log, error) {
+	return withObservedQueryAndResults(o, queryName, func() ([]Log, error) {
+		return o.ORM.FilteredLogs(ctx, filter, limitAndSort, queryName)
+	})
+}
+
+func (o *ObservedORM) GetLatestBlock(ctx context.Context) (int64, error) {
+	return withObservedQuery(o, "GetLatestBlack", func() (int64, error) {
+		return o.ORM.GetLatestBlock(ctx)
+	})
+}
+
+func withObservedQueryAndResults[T any](o *ObservedORM, queryName string, query func() ([]T, error)) ([]T, error) {
```

Comment:
I guess this is the same as evm LogPoller, but just noticing it's kind of strange that these with methods accept `o *ObservedORM` as their first param instead of being implemented as methods of ObservedORM.

I'm okay with leaving it as-is though, or changing it

Reply:
(although I think that would still be possible either way?)

Reply:
Let's leave this out of scope for this PR. But I'll keep this in mind!